crates/bdk/src/wallet/mod.rs

```diff
@@ -94,6 +94,91 @@ pub struct Wallet<D = ()> {
     secp: SecpCtx,
 }
 
+/// A structure to update [`KeychainTxOutIndex`], [`TxGraph`] and [`LocalChain`] atomically.
+///
+/// [`LocalChain`]: local_chain::LocalChain
+#[derive(Debug, Clone)]
+pub struct WalletUpdate<K, A> {
+    /// Contains the last active derivation indices per keychain (`K`), which is used to update the
+    /// [`KeychainTxOutIndex`].
+    pub last_active_indices: BTreeMap<K, u32>,
+
+    /// Update for the [`TxGraph`].
+    pub graph: TxGraph<A>,
+
+    /// Update for the [`LocalChain`].
+    ///
+    /// [`LocalChain`]: local_chain::LocalChain
+    pub chain: Option<local_chain::Update>,
+}
+
+impl<K, A> WalletUpdate<K, A> {
+    /// Construct a [`WalletUpdate`] with a given [`local_chain::Update`].
+    pub fn new(chain_update: local_chain::Update) -> Self {
+        Self {
+            last_active_indices: BTreeMap::new(),
+            graph: TxGraph::default(),
+            chain: Some(chain_update),
+        }
+    }
```

Comment:
Since the `chain` field is now an `Option`, we can implement `Default` for `WalletUpdate` instead of this method.

Don't use `derive()`, otherwise we will need to have `K` and `A` implement `Default` for `WalletUpdate` to be `Default`.

Reply:
Let me do this